File: php/class-access/sub/Value.php
<?php

namespace sub;

// NOTE: どこからでも利用可能
class Value
{
    // NOTE: Processor から参照させるためには public にするしかない
    //       その結果、他のどのクラスからも参照可能になってしまう
    public $value;

    public function __construct(int $value)
    {
        $this->value = $value;
    }
}

File: java/class-access/sub/Value.java
package sub;

// NOTE: パッケージ外から利用不可
class Value {
    // NOTE: 同一パッケージの Processor からは参照可能
    //       パッケージの外からは参照不可
    int value;

    Value(int value) {
        this.value = value;
    }
}
